src/LinkDotNet.NCronJob/RetryPolicies/FixedIntervalRetryPolicyCreator.cs

```diff
@@ -0,0 +1,34 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Polly;
+
+namespace LinkDotNet.NCronJob;
+
+/// <summary>
+/// A policy creator that configures a fixed interval retry policy.
+/// </summary>
+public partial class FixedIntervalRetryPolicyCreator : IPolicyCreator, IInitializablePolicyCreator
+{
+#pragma warning disable CS8618
+    private ILogger<FixedIntervalRetryPolicyCreator> logger;
+
+    /// <inheritdoc />
+    public void Initialize(IServiceProvider serviceProvider) =>
```

Comment:
Does it need the `Initialize` around?

The logger can be crated as ctor dependency?

Reply:
That class can't use ctor because it's used in the RetryPolicyAttribute<T> attribute as a generic parameter. That's why I had to add this initialization pattern so I could access the IServiceProvider.

Reply:
Fair!